app/views/components/_contents-list.html.erb

```diff
@@ -20,8 +20,12 @@
         <li
           class="app-c-contents-list__list-item app-c-contents-list__list-item--<%= parent_list_item_modifier %>"
         >
-          <%= link_to contents_item[:href], class: 'app-c-contents-list__link' do %>
-            <%= format_numbers ? wrap_numbers_with_spans(contents_item[:text]) : contents_item[:text] %>
+          <% if contents_item[:href] == nil %>
+            <%= contents_item[:text] %>
+          <% else %>
+            <%= link_to contents_item[:href], class: 'app-c-contents-list__link' do %>
```

Comment:
I think you can also use [link_to_if](https://apidock.com/rails/ActionView/Helpers/UrlHelper/link_to_if):

```rb
<%= link_to_if contents_item[:href], contents_item[:href], class: 'app-c-contents-list__link' do %>
  <%= format_numbers ? wrap_numbers_with_spans(contents_item[:text]) : contents_item[:text] %>
<% end %>
```